api_calls.py

```diff
@@ -1,81 +1,234 @@
 import http.client
 import json
 import datetime
+import time
 
 api_key = '6musfbnhsy2dembz5cxzxade'
-connection = http.client.HTTPSConnection("api.sportradar.us")
-
 month = datetime.datetime.now().month
 day = datetime.datetime.now().day
 year = datetime.datetime.now().year
 
 
 
-# Get the current schedule of games for today
-def get_current_schedule():
-    connection.request("GET", f"/nba/trial/v8/en/games/{year}/{month}/{day}/schedule.json?api_key={api_key}")
-    response = connection.getresponse()
-    data = response.read()
+## Facade Pattern for the API calls
+class SportsAPI:
+    ## Constructor for the class
+    def __init__(self):
+        self.api_key = api_key
+        self.connection = http.client.HTTPSConnection("api.sportradar.us")
+
+    def get_game_id(self, team_name, year, month, day) -> int:
+        game_id = None
+
+        try:
+            ## Request to API for the game id
+            self.connection.request("GET", f"/nba/trial/v8/en/games/{year}/{month}/{day}/schedule.json?api_key={self.api_key}")
+            response = self.connection.getresponse()
+
+            if response.status != 200:
+                print("Error: ", response.status, response.reason)
+                return None
+            
+            data = response.read()
+            json_data = json.loads(data.decode("utf-8"))
+
+            # Loop through the games and find the game id the team name
+            for game in json_data['games']:
+                if game['home']['name'] == team_name:
+                    game_id = game['id']
+                elif game['away']['name'] == team_name:
+                    game_id = game['id']
+
+        ## Catching exceptions
+        except json.JSONDecodeError as e:
+            print(f"A JSONDecodeError occurred: {str(e)}")
+        except http.client.HTTPException as e:
+            print(f"An exception occurred: {str(e)}")
+        except Exception as e:
+            print(f"An exception occurred: {str(e)}")
+        return game_id
+    
 
-    try:
-        json_data = json.loads(data.decode("utf-8"))
+    def get_team_id(self, team_name) -> int:
+        team_id = None
+
+        try:
+            ## Request to API for the team id
+            self.connection.request("GET", f"/nba/trial/v8/en/games/{year}/{month}/{day}/schedule.json?api_key={self.api_key}")
+            response = self.connection.getresponse()
+
+            if response.status != 200:
+                print("Error: ", response.status, response.reason)
+                return None
+            
+            data = response.read()
+            json_data = json.loads(data.decode("utf-8"))
+
+            # Loop through the games and find the team id from param team_name
+            for game in json_data['games']:
+                if game['home']['name'] == team_name:
+                    team_id = game['home']['id']
```

Comment:
I realize this is nitpicky, but I think this could be replaced with a return statement.